docs/ADR-42-third-party-collections-registry.md

```diff
@@ -0,0 +1,86 @@
+# Third Party Collections Registry
+
+## Statement of the problem
+
+Multiple third parties with their own NFT contracts (ERC721, ERC1155, etc) want to be part of Decentraland, but the current Decentraland collections implementations are not always a good fit for their use cases. Therefore, a smart contract is going to be created to have a decentralized way where they can map 3d assets to their already created NFTs.
+
+## Contract Implementation
+
+The TPR smart contract is going to be deployed on Polygon and support meta-transactions with the EIP-712 to reduce operational costs. Also, the contract can be easily migrate because it doesn't store or mint any token. The main purposes of this registry is to have an on-chain way to check whether an item has been approved by a committee member and therefore can be submitted to the Decentraland catalysts. And, to check whether a third party or item has been approved or rejected.
+
+The contract is not a storage-gas-consumption top efficient because it is priorize a way to loop through third parties and items without the need of indexing historical data.
```

Comment:
```suggestion
The contract is not a storage-gas-consumption top efficient because it prioritizes looping through third parties and items without the need of indexing historical data.
```